zanata-war/src/main/java/org/zanata/action/LanguageJoinAction.java

```diff
@@ -156,6 +203,60 @@ public void send() {
         }
     }
 
+    public boolean isUserAlreadyRequest() {
+        return requestServiceImpl.doesLanguageRequestExist(authenticatedAccount,
+            getLocale());
+    }
+
+    public void cancelRequest() {
+        LanguageRequest languageRequest =
+                requestServiceImpl.getPendingLanguageRequests(
+                        authenticatedAccount,
+                        getLocale().getLocaleId());
+        if(languageRequest == null) {
+            facesMessages.addGlobal(msgs.get("jsf.language.request.processed"));
+            return;
+        }
+
+        String comment =
+                "Request cancelled by requester {"
+                        + authenticatedAccount.getUsername() + "}";
+        requestServiceImpl.updateLanguageRequest(languageRequest.getId(),
+                authenticatedAccount, RequestState.CANCELLED, comment);
+
+        facesMessages.addGlobal(msgs.format("jsf.language.request.cancelled",
+            authenticatedAccount.getUsername()));
+    }
+
+    public String getMyRoles() {
+        if(authenticatedAccount == null) {
+            return "";
+        }
+        HLocaleMember localeMember = getLocaleMember();
+        if(localeMember == null) {
+            return "";
+        }
+
+        if(localeMember.isCoordinator()) {
+            return msgs.format("jsf.language.myRole",
+                    StringUtils.lowerCase(msgs.get("jsf.Coordinator")));
+        }
+
+        List<String> roles = Lists.newArrayList();
+        if(localeMember.isTranslator()) {
+            roles.add(StringUtils.lowerCase(msgs.get("jsf.Translator")));
+        }
+
+        if(localeMember.isReviewer()) {
+            roles.add(StringUtils.lowerCase(msgs.get("jsf.Reviewer")));
+        }
+
+        if(roles.size() == 1) {
+            return msgs.format("jsf.language.myRole", Joiner.on(" and ").join(roles));
+        }
+        return msgs.format("jsf.language.myRoles", Joiner.on(" and ").join(roles));
```

Comment:
This isn't translation safe because it assumes English plural rules, plus it concatenates hard-coded the English word "and" according to English grammar.

I suggest something like 

    jsf.language.myRoles:You have the following role/roles in this team: {0}

where {0} would be [admin] or [translator, coordinator].

Using "role/roles" is ugly, but at least it's consistently ugly in many languages. To fix it properly, we would need to use a different i18n approach, one which understands plurals. I think that's out of scope here.